### ISSUE TITLE
`controller` module returns empty inputs for Gamecube controller

### ISSUE THREAD

**Issue by ComplexPlane** [Felk/dolphin#10]
When calling `controller.get_gc_buttons()`, all button inputs appear to be `False` regardless of which buttons are pressed. The `Connected` field seems to be `True` though. I'm testing with a Gamecube controller plugged into port 0 into a Wii U USB adapter.

**Comment by Felk** [comment]
@ComplexPlane can you try this build please? I don't have an USB adapter to test.
[dolphin-python-scripting-gc-wii-usb-adapter-fix.zip](https://github.com/Felk/dolphin/files/8326396/dolphin-python-scripting-gc-wii-usb-adapter-fix.zip)

Looking at the Netplay code it seems necessary to explicitly handle controllers connected through an USB adapter. For future reference:
```patch
Index: Source/Core/Core/API/Controller.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Source/Core/Core/API/Controller.cpp b/Source/Core/Core/API/Controller.cpp
--- a/Source/Core/Core/API/Controller.cpp	(revision 46b7eacd5c810c2d21ec5fe51ea1a9c61a7ceb3d)
+++ b/Source/Core/Core/API/Controller.cpp	(date 1647967888854)
@@ -5,7 +5,10 @@
 #include "Controller.h"
 
 #include "Core/HW/GCPad.h"
+#include "Core/HW/SI/SI_Device.h"
 #include "Core/HW/Wiimote.h"
+#include "Core/ConfigManager.h"
+#include "InputCommon/GCAdapter.h"
 
 namespace API
 {
@@ -15,7 +18,10 @@
   auto iter = m_overrides.find(controller_id);
   if (iter != m_overrides.end())
     return iter->second.pad_status;
-  return Pad::GetStatus(controller_id);
+  if (SConfig::GetInstance().m_SIDevice[controller_id] == SerialInterface::SIDEVICE_WIIU_ADAPTER)
+    return GCAdapter::Input(controller_id);
+  else
+    return Pad::GetStatus(controller_id);
 }
 
 void GCManip::Set(GCPadStatus pad_status, int controller_id, ClearOn clear_on)
```
There's even a third case for GBA but I ignored that for now.

**Comment by Felk** [comment]
I'll assume that I fixed this in ce3b17811156fad408d60bdb844b9fbe95f5eea2
